pennylane/qchem/openfermion_obs.py

```diff
@@ -919,22 +919,32 @@ def molecular_hamiltonian(
     + (0.176276408043196) [Z2 Z3]
     """
     if len(coordinates) == len(symbols) * 3:
-        geometry_dhf = coordinates.reshape(len(symbols), 3)
+        geometry_dhf = qml.numpy.array(coordinates.reshape(len(symbols), 3))
         geometry_hf = coordinates
     elif len(coordinates) == len(symbols):
-        geometry_dhf = coordinates
+        geometry_dhf = qml.numpy.array(coordinates)
         geometry_hf = coordinates.flatten()
 
     if method == "dhf":
-        if args is None and type(geometry_dhf) is qml.numpy.tensor:
+        if args is None and isinstance(geometry_dhf, qml.numpy.tensor):
```

Comment:
Linting: although unrelated, likely Codefactor would fail due to changes in this file, hence this line is updated.